crates/services/txpool/src/txpool.rs

```diff
@@ -535,9 +542,11 @@ where
             .check_signatures(&consensus_params.chain_id())?;
 
         let parameters = CheckPredicateParams::from(consensus_params);
-        let tx =
-            tokio_rayon::spawn_fifo(move || tx.check_predicates(&parameters, memory))
-                .await?;
+        let tx = tokio_rayon::spawn_fifo(move || {
+            let view = provider.latest_view().unwrap();
```

Comment:
@xgreenx Instead of passing the whole provider and ask a view for each tx can we instead get a view before and reuse it ?

Reply:
Yes, it is better to get review and clone it to each transaction